### ISSUE TITLE
How to run from saved model

### ISSUE THREAD

**Issue by ajinkya933** [TheAILearner/A-CRNN-model-for-Text-Recognition-in-Keras#2]
@TheAILearner  I have trained and saved model "best_model.hdf5", how do I run from this saved model without rerunning the entire script

**Comment by TheAILearner** [comment]
Here is the script to run the code from trained model.
`# load the model
from keras.models import load_model
import numpy as np
import cv2
import keras.backend as K
import string

# char_list:   'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789'
# total number of our output classes: len(char_list)
char_list = string.ascii_letters+string.digits
 
def encode_to_labels(txt):
    # encoding each output word into digits
    dig_lst = []
    for index, char in enumerate(txt):
        try:
            dig_lst.append(char_list.index(char))
        except:
            print(char)
        
    return dig_lst

act_model = load_model('crnn_model.h5')

#test image path
test_img_path = r'mjsynth.tar\mnt\ramdisk\max\90kDICT32px\2852\4\15_BELLY_6876.jpg'

img = cv2.cvtColor(cv2.imread(test_img_path), cv2.COLOR_BGR2GRAY)   
 
# convert each image of shape (32, 128, 1)
w, h = img.shape
if h > 128 or w > 32:
    pass
if w < 32:
    add_zeros = np.ones((32-w, h))*255
    img = np.concatenate((img, add_zeros))

if h < 128:
    add_zeros = np.ones((32, 128-h))*255
    img = np.concatenate((img, add_zeros), axis=1)
img = np.expand_dims(img , axis = 2)
img = np.expand_dims(img , axis = 0)

# Normalize each image
img = img/255.

# predict outputs on validation images
prediction = act_model.predict(img)
 
# use CTC decoder
out = K.get_value(K.ctc_decode(prediction, input_length=np.ones(prediction.shape[0])*prediction.shape[1],
                         greedy=True)[0][0])
 
# see the results
i = 0
for x in out:
    print("original_text = ", test_img_path.split('_')[1])
    print("predicted text = ", end = '')
    for p in x:  
        if int(p) != -1:
            print(char_list[int(p)], end = '')       
    print('\n')
    i+=1`

**Comment by Arunpoochelvan** [comment]
import os
import fnmatch
import cv2
import numpy as np
import string
import time

from keras.preprocessing.sequence import pad_sequences

from keras.layers import Dense, LSTM, Reshape, BatchNormalization, Input, Conv2D, MaxPool2D, Lambda, Bidirectional
from keras.models import Model
from keras.activations import relu, sigmoid, softmax
import keras.backend as K
from keras.utils import to_categorical
from keras.callbacks import ModelCheckpoint

char_list = string.ascii_letters+string.digits

inputs = Input(shape=(32,128,1))
 
# convolution layer with kernel size (3,3)
conv_1 = Conv2D(64, (3,3), activation = 'relu', padding='same')(inputs)
# poolig layer with kernel size (2,2)
pool_1 = MaxPool2D(pool_size=(2, 2), strides=2)(conv_1)
 
conv_2 = Conv2D(128, (3,3), activation = 'relu', padding='same')(pool_1)
pool_2 = MaxPool2D(pool_size=(2, 2), strides=2)(conv_2)
 
conv_3 = Conv2D(256, (3,3), activation = 'relu', padding='same')(pool_2)
 
conv_4 = Conv2D(256, (3,3), activation = 'relu', padding='same')(conv_3)
# poolig layer with kernel size (2,1)
pool_4 = MaxPool2D(pool_size=(2, 1))(conv_4)
 
conv_5 = Conv2D(512, (3,3), activation = 'relu', padding='same')(pool_4)
# Batch normalization layer
batch_norm_5 = BatchNormalization()(conv_5)
 
conv_6 = Conv2D(512, (3,3), activation = 'relu', padding='same')(batch_norm_5)
batch_norm_6 = BatchNormalization()(conv_6)
pool_6 = MaxPool2D(pool_size=(2, 1))(batch_norm_6)
 
conv_7 = Conv2D(512, (2,2), activation = 'relu')(pool_6)
 
squeezed = Lambda(lambda x: K.squeeze(x, 1))(conv_7)
 
# bidirectional LSTM layers with units=128
blstm_1 = Bidirectional(LSTM(128, return_sequences=True, dropout = 0.2))(squeezed)
blstm_2 = Bidirectional(LSTM(128, return_sequences=True, dropout = 0.2))(blstm_1)
 
outputs = Dense(len(char_list)+1, activation = 'softmax')(blstm_2)

# model to be used at test time
act_model = Model(inputs, outputs)


act_model.load_weights('crnn_model.h5')
 
img = cv2.cvtColor(cv2.imread("fine.png"), cv2.COLOR_BGR2GRAY)   

# convert each image of shape (32, 128, 1)
w, h = img.shape

if w < 32:
    add_zeros = np.ones((32-w, h))*255
    img = np.concatenate((img, add_zeros))

if h < 128:
    add_zeros = np.ones((32, 128-h))*255
    img = np.concatenate((img, add_zeros), axis=1)
img = np.expand_dims(img , axis = 2)

# Normalize each image
img = img/255.
valid_img = np.array(img)

valid_img = valid_img[np.newaxis, ...]
# predict outputs on validation images
prediction = act_model.predict(valid_img)
 
# use CTC decoder   
out = K.get_value(K.ctc_decode(prediction, input_length=np.ones(prediction.shape[0])*prediction.shape[1],
                         greedy=True)[0][0])
 
# see the results
i = 0
for x in out:
    print("original_text =  seconal")
    print("predicted text = ", end = '')
    for p in x:  
        if int(p) != -1:
            print(char_list[int(p)], end = '')       
    print('\n')
    i+=1

**Comment by neenaloysius** [comment]
When I run CTC_decoder im getting all -1 in out. Plz could you clarify why im getting -1??
